src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -148,8 +155,48 @@ public async Task StartWorkerProcessAsync()
 
         public async Task<WorkerStatus> GetWorkerStatusAsync()
         {
+            return await GetWorkerStatusAsync(true);
+        }
+
+        public WorkerStatus GetWorkerStatus()
+        {
+            return new WorkerStatus()
+            {
+                ProcessStats = _rpcWorkerProcess.GetStats(),
+                RpcWorkerStats = _monitor.GetStats(),
+            };
+        }
+
+        public async Task<WorkerStatus> GetWorkerStatusAsync(bool sendWorkerStatusResquest)
+        {
+            // This path uses Sacale controller call "admin/host/ping"
             var workerStatus = new WorkerStatus();
 
+            TimeSpan? workerStatusLatency = await GetWorkerStatusResponseAsync(null);
+            if (workerStatusLatency != null)
+            {
+                workerStatus.Latency = workerStatusLatency.Value;
+                _workerChannelLogger.LogDebug($"[HostMonitor] Worker status request took {workerStatusLatency.Value.TotalMilliseconds}ms");
+            }
+
+            // get the process stats for the worker
+            var workerProcessStats = _rpcWorkerProcess.GetStats();
```

Comment:
WorkerStatus.ProcessStats (including the ProcessStats class itself) are going away in https://github.com/Azure/azure-functions-host/pull/7406. All process monitoring for the host is now done in the WebJobs SDK, and child process stats are already logged there (code [here](https://github.com/Azure/azure-webjobs-sdk/blob/7bfcd07e520210c9cc092e6c1ac9413fc5c3d178/src/Microsoft.Azure.WebJobs.Host/Scale/DefaultHostProcessMonitor.cs#L103)). Those states are also feeding into ConcurrencyManager throttle decisions.

So you should remove any usages of ProcessStats. Likely, this PR will need to go in after mine - we'll need to coordinate.

Reply:
I will update the PR once #7406 is merged.

Reply:
This is updated.